src/super_gradients/training/sg_model/sg_model.py

```diff
@@ -320,6 +313,91 @@ def build_model(self,  # noqa: C901 - too complex
                 self.arch_params.num_classes = num_classes_new_head
                 self.net.to(self.device)
 
+    def build_kd_model(self,
```

Comment:
@shaydeci  - Something does not feel right...
All of the checkpoints mechanism for KD needs to be polished and adapted accordingly...

BTW - I suggest to have all of the code that handles KD in the same place, and not spanned across all of our code base... namely - SG_Model SHOULD NOT BE CHANGED in order to support an extended use case... We should inherit from sg_model and implement kd_model that overrides the build_model method according to the required behaviour...

This is just one use case, but this is relevant for all of the required changes.. like save_checkpoint, changes to the metrics or anything else that is *specific* to a KDModel and not relevant for an SGModel...

We need to be more aware of this design since we will eventually want to take apart the SGModel monolith into smaller modules and we do not want to feed the monster